content/documentation/managing_the_server/security.md

```diff
@@ -104,25 +55,160 @@ Examples using the test certicates which are self signed for localhost and 127.0
 ```
 
+This option simply verifies the client's certificate has been signed by the CA specified in the `ca_file` option. However, it does not map any attribute of the client's certificate to the user's identity.
+
+To have TLS Mutual Authentication map certificate attributes to the users identity, replace the option `verify` with `verify_and_map` as shown as follows:
+
+```ascii
+tls {
+  cert_file: "./configs/certs/server-cert.pem"
+  key_file:  "./configs/certs/server-key.pem"
+  ca_file:   "./configs/certs/ca.pem"
+  # Require a client certificate and map user id from certificate
+  verify_and_map: true
+}
+```
+
+There are two options for certificate attributes that can be mapped to user names. The first is the email address in the Subject Alternative Name (SAN) field of the certificate. While generating a certificate with this attribute is outside the scope of this document, we will view this it with OpenSSL:
+
+```ascii
+$ openssl x509 -noout -text -in  test/configs/certs/client-id-auth-cert.pem
+Certificate:
+  -------------<truncated>-------------
+        X509v3 extensions:
+            X509v3 Subject Alternative Name:
+                DNS:localhost, IP Address:127.0.0.1, email:derek@nats.io
+            X509v3 Extended Key Usage:
+                TLS Web Client Authentication
+  -------------<truncated>-------------
+```
+
+The configuration to authorize this user would be as follows:
+
+```ascii
+authorization {
+  users = [
+    {user: "derek@nats.io", permissions: { publish: "foo" }}
+  ]
+}
+```
+
+Note: This configuration only works for the first email address if there are multiple emails in the SAN field.
+
+The second option is to use the RFC 2253 Distinguished Names syntax from the certificate subject as follows:
+
+```ascii
+$ openssl x509 -noout -text -in  test/configs/certs/tlsauth/client2.pem
+Certificate:
+    Data:
+  -------------<truncated>-------------
+        Subject: OU=CNCF, CN=example.com
+  -------------<truncated>-------------
+```
+
+The configuration to authorize this user would be as follows:
+
+```ascii
+authorization {
+  users = [
+    {user: "CN=example.com,OU=CNCF", permissions: { publish: "foo" }}
+  ]
+}
+```
+
+### Cluster TLS Mutual Authentication
+
+When setting up clusters all servers in the cluster, if using TLS, will both verify the connecting endpoints and the server responses. So certificates are checked in both directions. Certificates can be configured only for the server's cluster identity, keeping client and server certificates separate from cluster formation.
+
+```ascii
+cluster {
+  listen: 127.0.0.1:4244
+
+  tls {
+    # Route cert
+    cert_file: "./configs/certs/srva-cert.pem"
+    # Private key
+    key_file:  "./configs/certs/srva-key.pem"
+    # Optional certificate authority verifying connected routes
+    # Required when we have self-signed CA, etc.
+    ca_file:   "./configs/certs/ca.pem"
```

Comment:
btw `verify_and_map` is also supported in this section but only with a single user.